libevm/triedb/firewood/account_trie.go

```diff
@@ -0,0 +1,299 @@
+// Copyright 2025 the libevm authors.
+//
+// The libevm additions to go-ethereum are free software: you can redistribute
+// them and/or modify them under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation, either version 3 of the License,
+// or (at your option) any later version.
+//
+// The libevm additions are distributed in the hope that they will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser
+// General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see
+// <http://www.gnu.org/licenses/>.
+
+package firewood
+
+import (
+	"errors"
+
+	"github.com/ava-labs/libevm/common"
+	"github.com/ava-labs/libevm/core/state"
+	"github.com/ava-labs/libevm/core/types"
+	"github.com/ava-labs/libevm/crypto"
+	"github.com/ava-labs/libevm/ethdb"
+	"github.com/ava-labs/libevm/log"
+	"github.com/ava-labs/libevm/rlp"
+	"github.com/ava-labs/libevm/trie"
+	"github.com/ava-labs/libevm/trie/trienode"
+	"github.com/ava-labs/libevm/triedb/database"
+)
+
+var _ state.Trie = (*AccountTrie)(nil)
+
+// AccountTrie implements state.Trie for managing account states.
+// There are a couple caveats to the current implementation:
+//  1. `Commit` is not used as expected in the state package. The `StorageTrie` doesn't return
+//     values, and we thus rely on the `AccountTrie`.
+//  2. The `Hash` method actually creates the proposal, since Firewood cannot calculate
+//     the hash of the trie without committing it. It is immediately dropped, and this
+//     can likely be optimized.
+//
+// Note this is not concurrent safe.
+type AccountTrie struct {
+	fw           *Database
+	parentRoot   common.Hash
+	root         common.Hash
+	reader       database.Reader
+	dirtyKeys    map[string][]byte // Store dirty changes
```

Comment:
You could use `sync.Map` as a direct replacement or add a `sync.RWMutex` to guard access to the map.

Also consider if sharing `AccountTrie` instances across goroutines is necessary, or if each goroutine should have its own copy.